stock_intercompany/tests/test_intercompany_picking.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2021 Camptocamp
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests.common import TransactionCase
+
+
+class TestIntercompanyDelivery(TransactionCase):
+    def setUp(self):
+        super().setUp()
+        self.user_demo = self.env.ref("base.user_demo")
+        company_obj = self.env["res.company"]
+        # Create 2 companies and configure intercompany picking type param on them
+        self.company1 = company_obj.create({"name": "Company A"})
+        self.company2 = company_obj.create({"name": "Company B"})
+        self.picking_type_1 = (
+            self.env["stock.picking.type"]
+            .sudo()
+            .search(
+                [
+                    ("company_id", "=", self.company1.id),
+                    ("name", "=", "Delivery Orders"),
+                ],
+                limit=1,
+            )
+        )
+        self.picking_type_2 = (
+            self.env["stock.picking.type"]
+            .sudo()
+            .search(
+                [("company_id", "=", self.company2.id), ("name", "=", "Receipts")],
+                limit=1,
+            )
+        )
+
+        self.company1.intercompany_in_type_id = self.picking_type_1.id
+        self.company2.intercompany_in_type_id = self.picking_type_2.id
+        # assign both companies to current user
+        self.user_demo.write(
+            {
+                "company_id": self.company1.id,
+                "company_ids": [(4, self.company1.id), (4, self.company2.id)],
+            }
+        )
+        # create storable product
+        self.product1 = self.env["product.product"].create(
+            {
+                "name": "Product A",
+                "type": "product",
+                "categ_id": self.env.ref("product.product_category_all").id,
+                "qty_available": 100,
```

Comment:
don't understand why checking product with default_company_id inside with_context(...) shows qty_available -1.0 for both company1 and company2. with this not possible to make a  assert and check product quantities on both companies when pickings are created.